### DISCUSSION TITLE
Support large TMX maps

### DISCUSSION THREAD

**Discussion by DeathPhoenix22** [AlmasB/FXGL#1304] [Ideas]
Hi,

I'm currently facing a limitation with the TilesetLoader. When the map is too large, we run out of memory. This is due to the fact that we load the map in a single pixelsBuffer as 1 image.

Here is what I would suggest:
- Acquire the Prism (GPU) memory size configured and ensure that no pixel buffers exceeds it
- Load TMX maps in segments based on need (That one will be extra challenging)
- Automatically support Zoom caching so we can use smaller versions of the map segments when zoomed out (this will have a great impact on both Memory usage and draw speed)
- Automatically cleanse memory as we cannot keep all pixels buffers we load forever.

There is another memory issue related to the TMX loading sequence. In fact, we load textures in a separated cache (directly in the TilesetLoader), which means that all Levels related textures will most likely be stored 2x in memory. I'll try to address that, but it could be a problem too.

Is that a good plan?
Any suggestions?

Thanks,

**Comment by AlmasB** [comment]
These are some good points. It would be good to start with an example (map + texture) that fails to load due to above. We can then turn this into a test and finally figure out a suitable solution.
